src/main/java/com/savitoh/casacodigoapi/validator/AutorCustomValidator.java

```diff
@@ -0,0 +1,30 @@
+package com.savitoh.casacodigoapi.validator;
+
+import com.savitoh.casacodigoapi.exception.data.CreateAutorValidateException;
+import com.savitoh.casacodigoapi.model.Autor;
+import com.savitoh.casacodigoapi.repository.AutorRepository;
+import org.springframework.stereotype.Component;
+
+import java.util.Objects;
+
+@Component
+public class AutorCustomValidator  {
```

Comment:
*Deixamos pistas que facilitem o uso do código onde não conseguimos resolver com compilação.*

Se aqui você verifica autor com mesmo email, você poderia chamar de AutorComMesmoEmailValidator :). 

Ah, no card até tava liberado email duplicado :). Mas legal fazer a validação já aqui.. Vai facilitar sua vida quando for salvar uma categoria.

Reply:
De fato ficou um nome muito genérico.